app/src/androidTest/java/com/github/ybecker/epforuml/NavigationTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.ybecker.epforuml
+
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class NavigationTest {
+
+    @Test
+    fun testFragmentNavigation() {

```

Comment:
You could improve this test by splitting it into smaller parts, so that if it fails, you can easily identify which tests are failing and find your errors in your code more efficiently.